metapool/sample_sheet.py

```diff
@@ -803,6 +815,106 @@ def __init__(self, path=None):
+        'IEMFileVersion': '4',
+        'SheetType': _STANDARD_METAG_SHEET_TYPE,
+        'SheetVersion': '101',
+        'Investigator Name': 'Knight',
+        'Experiment Name': 'RKL_experiment',
+        'Date': None,
+        'Workflow': 'GenerateFASTQ',
+        'Application': 'FASTQ Only',
+        'Assay': _METAGENOMIC,
+        'Description': '',
+        'Chemistry': 'Default',
+    }
+
+    data_columns = ['Sample_ID', 'Sample_Name', 'Sample_Plate', 'well_id_384',
+                    'I7_Index_ID', 'index', 'I5_Index_ID', 'index2',
+                    'Sample_Project', 'Well_description']
+
+    # columns present in an pre-prep file (amplicon) that included katharoseq
+    # controls. Presumably we will need these same columns in a sample-sheet.
+    optional_katharoseq_columns = ['Kathseq_RackID', 'TubeCode',
+                                   'katharo_description',
+                                   'number_of_cells',
+                                   'platemap_generation_date',
+                                   'project_abbreviation',
+                                   'vol_extracted_elution_ul', 'well_id_96']
+
+    # For now, assume only MetagenomicSampleSheetv101 (v100, v95, v99) contains
+    # 'contains_replicates' column. Assume AbsQuantSampleSheetv10 doesn't.
+    _BIOINFORMATICS_COLUMNS = {'Sample_Project', 'QiitaID', 'BarcodesAreRC',
+                               'ForwardAdapter', 'ReverseAdapter',
+                               'HumanFiltering', 'contains_replicates',
+                               'library_construction_protocol',
+                               'experiment_design_description'}
+
+    CARRIED_PREP_COLUMNS = ['experiment_design_description', 'i5_index_id',
+                            'i7_index_id', 'index', 'index2',
+                            'library_construction_protocol', 'sample_name',
+                            'sample_plate', 'sample_project',
+                            'well_description', 'well_id_384']
+
+    def __init__(self, path=None):
+        super().__init__(path=path)
+        self.remapper = {
+            'sample sheet Sample_ID': 'Sample_ID',
+            'Sample': 'Sample_Name',
+            'Project Plate': 'Sample_Plate',
+            'Well': 'well_id_384',
+            'i7 name': 'I7_Index_ID',
+            'i7 sequence': 'index',
+            'i5 name': 'I5_Index_ID',
+            'i5 sequence': 'index2',
+            'Project Name': 'Sample_Project',
+            'Kathseq_RackID': 'Kathseq_RackID'
+        }
+
+    def contains_katharoseq_samples(self):
+        # when creating samples manually, as opposed to loading a sample-sheet
+        # from file, whether or not a sample-sheet contains katharoseq
+        # controls can change from add_sample() to add_sample() and won't be
+        # determined when MetagenomicSampleSheetv101() is created w/out a
+        # file. Hence, perform this check on demand() as opposed to once at
+        # init().
```

Comment:
I would prefer katharo as well, but I based it on what I saw in @cbrenchy's PR and some legacy sample-sheets. That's the main reason I asked the others to also review the PR, but as of yet I've heard no comment either way. I think it's suitable to run with, at least for the time being. 

lower-case because Caitriona has used lower-cased 'kath' and we've used upper-case 'KATH' in the past.

Reply:
Sorry for delayed response, I was out of office - katharo is preferred. It doesn't matter if we choose upper or lower case, but we need to stick to one for consistency.

Reply:
Great, thank you! Just to confirm, we would need to make changes in this PR and the notebooks to support katharo vs. kath, right? If yes, @charles-cowart, do you want to do that here or in another PR?

Reply:
Also, just to confirm the katharo naming scheme is as follows: "katharo".abbreviated project name.plate #.Well ID.CellCounts

From what I am seeing the naming is "kathseq.18000.0.E11", which is missing plate# and abbreviated project name. Or am I looking in the wrong place?

Reply:
> Sorry for delayed response, I was out of office - katharo is preferred. It doesn't matter if we choose upper or lower case, but we need to stick to one for consistency.

No worries, thank you for your reply! You got it - katharo instead of kath.

Reply:
> Great, thank you! Just to confirm, we would need to make changes in this PR and the notebooks to support katharo vs. kath, right? If yes, @charles-cowart, do you want to do that here or in another PR?

I'd like that to be a separate PR, because we will need the changes here to help do that.

Reply:
@antgonza I'll issue changes for the naming scheme in the examples in a subsequent PR. I'll have to review recent work on the notebooks to confirm the changes there as well and I'll need this code to make the changes. I'd like to merge this before continuing.